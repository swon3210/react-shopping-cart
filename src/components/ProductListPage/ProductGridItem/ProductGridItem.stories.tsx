import ProductGridItem, { Props } from './ProductGridItem';

export default {
  title: 'Components/ProductListPage/ProductGridItem',
  component: ProductGridItem,
  argTypes: {},
};

const Template = (args: Props) => <ProductGridItem {...args} />;

export const Default = Template.bind({});

(Default as any).args = {
  name: '상품 이름',
  price: '18,900',
  thumbnail:
    'https://images.unsplash.com/photo-1589365278144-c9e705f843ba?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=634&q=80',
};
