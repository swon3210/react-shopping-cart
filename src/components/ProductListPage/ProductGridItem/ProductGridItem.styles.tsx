import styled from 'styled-components';

export const ProductGridItem = styled.div`
  width: 282px;
  color: ${({ theme }) => theme.TEXT_COLOR};
  cursor: pointer;
`;

export const ProductThumbnail = styled.img`
  width: 100%;
  height: 282px;
  object-fit: cover;
  object-position: center;
  margin-bottom: 16px;
`;

export const GridBottomWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 14px;
`;

export const GridTextWrapper = styled.div`
  display: flex;
  flex-direction: column;
`;

export const ProductName = styled.span`
  line-height: 22px;
  margin-bottom: 5px;
`;

export const ProductPrice = styled.span`
  font-size: 20px;
`;

export const CartIcon = styled.img``;
