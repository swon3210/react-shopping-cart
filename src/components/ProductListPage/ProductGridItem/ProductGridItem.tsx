import * as Styled from './ProductGridItem.styles';
import shoppingCartDarkSVG from '../../../assets/svgs/shopping-cart-dark.svg';

export interface Props {
  name: string;
  price: string;
  thumbnail: string;
  onClick?: () => void;
  onCartButtonClick?: () => void;
}

const ProductGridItem = ({ name, price, thumbnail, onClick, onCartButtonClick }: Props) => {
  const onCartIconClick = (event: React.MouseEvent<HTMLImageElement>) => {
    event.stopPropagation();
    onCartButtonClick?.();
  };

  return (
    <Styled.ProductGridItem onClick={onClick}>
      <Styled.ProductThumbnail src={thumbnail} />
      <Styled.GridBottomWrapper>
        <Styled.GridTextWrapper>
          <Styled.ProductName>{name}</Styled.ProductName>
          <Styled.ProductPrice>{price} Ïõê</Styled.ProductPrice>
        </Styled.GridTextWrapper>
        <Styled.CartIcon onClick={onCartIconClick} src={shoppingCartDarkSVG} />
      </Styled.GridBottomWrapper>
    </Styled.ProductGridItem>
  );
};

export default ProductGridItem;
