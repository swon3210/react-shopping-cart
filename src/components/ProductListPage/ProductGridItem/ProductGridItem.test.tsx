import ProductGridItem from './ProductGridItem';
import { render } from '@testing-library/react';

describe('ProductGridItem Component', () => {
  it('ProductGridItem Snapshot', () => {
    const productGridItemUtil = render(<ProductGridItem price="2000" name="이름" thumbnail=""></ProductGridItem>);
    expect(productGridItemUtil.asFragment()).toMatchSnapshot();
  });
});
