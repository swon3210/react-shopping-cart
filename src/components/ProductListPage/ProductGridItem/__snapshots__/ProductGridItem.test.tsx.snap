// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductGridItem Component ProductGridItem Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvaa grImBg"
  >
    <img
      class="sc-gsDJrp ivitjl"
      src=""
    />
    <div
      class="sc-dkPtyc hevdYk"
    >
      <div
        class="sc-hKwCoD eZrVPk"
      >
        <span
          class="sc-eCImvq cWdtSg"
        >
          이름
        </span>
        <span
          class="sc-jRQAMF lgyETR"
        >
          2000 원
        </span>
      </div>
      <img
        class="sc-gKckTs"
        src="shopping-cart-dark.svg"
      />
    </div>
  </div>
</DocumentFragment>
`;
